apps/api/src/services/sendgrid_handler.py

```diff
@@ -65,6 +75,17 @@ async def send_email(
     ...
 
 
+@overload
+async def send_email(
+    template_id: Literal[Template.CONFIRMATION_EMAIL],
+    sender_email: Tuple[str, str],
+    receiver_data: Iterable[ApplicationUpdatePersonalization],
```

Comment:
Issue: template doesn't match receiver data, might need to specify a union of the three templates.